src/Components/TypeAhead.jsx

```diff
@@ -0,0 +1,121 @@
+import React from "react";
+import "../TypeAhead.css";
+import Tag from "./Tag";
+
+export default class TypeAhead extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      suggestions: [],
+      text: "",
+      selectedTags: [],
+    };
+  }
+
+  onTextChange = (e) => {
+    const { items } = this.props;
+    let suggestions = [];
+    const value = e.target.value;
+
+    console.log("Input value:", value); // Debugging
+
+    if (value.length > 0) {
+      // prettier-ignore
+      const regex = new RegExp(`^${value}`, "i");
+
+      console.log("Items being filtered:", items); // Debugging
+
+      suggestions = items.filter((v) => regex.test(v)).sort();
+    }
+
+    this.setState(() => ({
+      suggestions,
+      text: value,
+    }));
+  };
+
+  suggestionSelected = (value) => {
+    this.setState(
```

Comment:
Wherever you took inspiration from is very outdated. The keyword this is not needed here, you can use the useState hook. Please refactor this whole file.